### PR TITLE
Submit answer flow

### PR DESCRIPTION
**Pull request by Geor23**

### CODE REVIEW

File: client/js/services/userService.js [hunk 1]
```diff
@@ -37,6 +37,31 @@ ClonageApp.service('userService', ['socket', '$sessionStorage', function(socket,
...
+    function submitVote(enteredAnswer) {
+        alert(enteredAnswer);
+         emitVote(enteredAnswer);
+    }
+
+    function emitAnswer(answer){
+        socket.emit('USER answer', {
+            playerId: $sessionStorage.userId,
+            answer: answer,
+            roomId: $sessionStorage.roomId
```
Comment (JamesAdamHughes) [inline client/js/services/userService.js]: can just use the services's user variable to access the user id and room id, there are more likely to be up to date and can't be tampered with by Scott Logic peeps

File: server/GameController.js [hunk 2]
```diff
@@ -98,13 +107,107 @@ module.exports = function(data) {
...
+		var currentRound = rounds[rounds.length-1];
+		currentRound.answers.push(ans);
+		updateHand(playerId, answer);
+
+		//check if everyone submitted
+		if (currentRound.answers.length === players.length) {
+			callback({
+				answers: currentRound.answers
+			});
+		}
```
Comment (JamesAdamHughes) [inline server/GameController.js]: maybe an else cause here, so we can track how many answer have been submitted, error handling etc

File: server/server.js [hunk 3]
```diff
@@ -307,6 +363,23 @@ module.exports = function(port, enableLogging) {
         });
     }
 
+    function broadcastoptions(room, event, data) {
```
Comment (JamesAdamHughes) [inline server/server.js]: nb write comment to make functionality clearer

File: client/js/config/routeConfig.js [hunk 4]
```diff
@@ -13,6 +13,9 @@ ClonageApp.config(['$routeProvider', '$locationProvider',
             .when('/question/', {
                 templateUrl: '/templates/question.html'
             })
+            .when('/vote/', {
+                templateUrl: '/templates/vote.html'
+            })
```
Comment (JamesAdamHughes) [inline client/js/config/routeConfig.js]: There is no vote.html in the pull request

File: client/js/controllers/mainController.js [hunk 5]
```diff
@@ -6,6 +6,15 @@ ClonageApp.controller("MainController", function($scope, userService, roomServic
     $scope.roomId = roomService.getRoomId;
     $scope.roundQuestion = gameService.getRoundQuestion;
     $scope.userHand = userService.getUserHand;
+    $scope.answers = gameService.getAnswers;
+
+    $scope.submitAnswer = function(enteredAnswer) {
+        userService.submitAnswer(enteredAnswer);
+    };
+
+    $scope.submitVote = function(enteredAnswer) {
```
Comment (JamesAdamHughes) [inline client/js/controllers/mainController.js]: Where is this called?

File: client/js/services/userService.js [hunk 6]
```diff
@@ -37,6 +37,31 @@ ClonageApp.service('userService', ['socket', '$sessionStorage', function(socket,
...
+        });
+    }
+
+    function emitVote(answer){
+        socket.emit('USER vote', {
+            playerId: $sessionStorage.userId,
+            answer: answer,
+            roomId: $sessionStorage.roomId
+        });
```
Comment (JamesAdamHughes) [inline client/js/services/userService.js]: same as emitAnswer

File: server/GameController.js [hunk 7]
```diff
@@ -98,13 +107,107 @@ module.exports = function(data) {
...
+
+	/*
+	add 50 points to player -> called on each vote
+	*/
+	var addPoints = function(playerId) {
+		console.log("added vote");
+		players.forEach (function(player){
+			if (player.uId === playerId ) {
+				player.points += 50 ;
+			}
```
Comment (JamesAdamHughes) [inline server/GameController.js]: Set variable POINTS_PER_VOTE = 50 at top of file, so we can easily change it

File: client/js/config/routeConfig.js (hunk 4 above)
Comment (Geor23) [inline client/js/config/routeConfig.js]: added it

File: client/js/controllers/mainController.js (hunk 5 above)
Comment (Geor23) [inline client/js/controllers/mainController.js]: vote.html -added it sorry!

**Pull request merged**